### ISSUE TITLE
Quagga mussel error?

### ISSUE THREAD

**Issue by GiselleSterry** [nbnuk/nbnatlas-issues#283]
No return for this species. 
Email from C Raper 3/7/17
"I've tried searching for both the vernacular name and the scientific
name (Dreissena bugensis) and neither appears in the Species Search. The
names are old enough to have been included in the taxonomy I sent when they
set up the NBN Atlas so it should all be there."

**Comment by GiselleSterry** [comment]
Hi





Email received today: Have you been able to resolve this issue at all?  I've just been sent the following link which shows the two taxa aren’t present on the NHM website either - I don’t know if that helps in your investigations:

http://www.nhm.ac.uk/our-science/data/uk-species/species/names.html?name.match=CONTAINS&name.value=Dreissena&rank=&group=&_recommendedOnly=on&_wellFormedOnly=on&showScientificNames=true

Kind regards
Carole

**Comment by sophiathirza** [comment]
The occurrences are in the dataset but the species name is unprocessed: https://records.nbnatlas.org/occurrences/search?q=data_resource_uid%3Adr827&fq=-species%3A*&fq=raw_taxon_name%3A%22Dreissena+bugensis%22

which is why it doesn't seem possible to search on them. I will investigate why the names were not processed.

**Comment by sophiathirza** [comment]
The dataset is here: https://registry.nbnatlas.org/public/show/dr827

**Comment by sophiathirza** [comment]
Email from ChrisR:

I think I forwarded that request on to you because Quagga mussel has been in
the UKSI for ages. The only potential issue is that it does seem to be set
to only be on a private list, which is not correct but I'm pretty sure the
Gateway ignored this flag. It's possible your developer might have filtered
on it in error but he has never really discussed with me what he has done so
I can't comment. If you can get him to get in touch then I can chat with
him. In the meantime I will switch the flag off but your developer needs to
check with me how he is using the UKSI data.

**Comment by sophiathirza** [comment]
I have written to Chris to ask for a new export of the UKSI.

**Comment by djtfmartin** [comment]
Just for notes, in the current DwC-A we produced for the Atlas based on an export of UKSI, the vernacular name was pointing to a taxon NHMSYS0020475523 which is missing from the taxa.csv either due to absence in source data or a bug in the transformation script.

```
server> /data/bie/import/uksi
$ cat vernacular.csv | grep Quagga
"NHMSYS0020475523","NHMSYS0021118213","","Quagga Mussel","en","preferred"
server> /data/bie/import/uksi

$ cat taxa.csv | grep NHMSYS0020475523
"NHMSYS0021118235","","NHMSYS0020475523","","Dreissena rostriformis bugensis","Andrusov, 1897","Subspecies","","synonym"
````

**Comment by sophiathirza** [comment]
In the current DwC-A. Quagga mussel points to the sub species (Dreissena rostriformis bugensis) via the acceptedNameUsageID field. Dreissena bugensis (tsk: NHMSYS0020475523) still does not exist in the DwC-A.

**Comment by sophiathirza** [comment]
I re-ran the script after swapping the column indexes for 'establishmentMeans' and 'redundant' in the Organism_Master reader. I had previously changed the column index for 'dateDeleted' from 18 to 19.

Dreissena bugensis (tvk: NHMSYS0020475523) is now in the taxa.csv file and Quagga mussel points correctly to this tvk.

When we get the latest version of the UKSI, we must check the column numbers before running the script. I don't know if it was something I did when I exported the tables from Access or how the csv files csv were originally supplied.
